### PR TITLE
Progress Dialog: Fix recursion and concurrency use of text updates

### PR DESCRIPTION
**Pull request by elad335**
* Fix use-after-free of m_text.c_str() by system_progress_dialog after the object of scoped_progress_dialog is freed.

### CODE REVIEW

File: rpcs3/Emu/system_progress.cpp [hunk 1]
```diff
@@ -36,18 +37,188 @@ namespace rsx::overlays
 	};
 } // namespace rsx::overlays
 
+scoped_progress_dialog::scoped_progress_dialog(std::string text) noexcept
+{
```
Comment (Megamouse) [inline rpcs3/Emu/system_progress.cpp:41]: Please move this to another file. It's way too large

Comment (elad335) [reply rpcs3/Emu/system_progress.cpp]: Done

File: rpcs3/Emu/system_progress.cpp [hunk 2]
```diff
@@ -36,18 +37,94 @@ namespace rsx::overlays
 	};
 } // namespace rsx::overlays
 
+scoped_progress_dialog::~scoped_progress_dialog() noexcept
```
Comment (Megamouse) [inline rpcs3/Emu/system_progress.cpp:40]: 🧐

Comment (elad335) [reply rpcs3/Emu/system_progress.cpp]: 🤣

File: rpcs3/Emu/scoped_progress_dialog.cpp [hunk 3]
```diff
@@ -0,0 +1,207 @@
+#include "system_progress.hpp"
```
Comment (Megamouse) [inline rpcs3/Emu/scoped_progress_dialog.cpp:1]:
```suggestion
#include "stdafx.h"
#include "system_progress.hpp"
```

File: rpcs3/Emu/scoped_progress_dialog.cpp [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+	// Which is technically the newest value to be inserted so it serves its logic
+	if (unmap_index != umax && unmap_index != m_text_index)
+	{
+		g_progr_text_queue[unmap_index].reset();
+	}
+
+	return *this;
+}
+
+
```
Comment (Megamouse) [inline rpcs3/Emu/scoped_progress_dialog.cpp:131]: newline

**Pull request merged**